src/app/@ansyn/menu-items/layers-manager/components/layers-collection/layer-collection.component.ts

```diff
@@ -33,18 +40,49 @@ import { SelectLayerAction, UnselectLayerAction } from '@ansyn/menu-items/layers
 	]
 })
 
-export class LayerCollectionComponent {
-	@Input() collection: LayersContainer;
+export class LayerCollectionComponent implements OnInit {
+	@Input() collection: Layer[];
+	_selectedLayers = [];
+	selectedLayers$: Observable<any> = this.store.select(selectedLayersIds)
+		.distinctUntilChanged()
+		.do((selectedLayers) => {
+			this._selectedLayers = selectedLayers;
+			this.collectionWithIsChecked = this.collectionWithIsChecked.map((layer) => {
+				return { ...layer, isChecked: this._selectedLayers.includes(layer.id) };
+			});
+		});
+
 	public show = true;
+	subscribers = [];
+	collectionWithIsChecked: any[];
+
+
+	ngOnInit() {
+
+		this.collectionWithIsChecked = cloneDeep(this.collection);
+
+		this.subscribers.push(
+			this.selectedLayers$.subscribe()
+		);
+	}
 
 	constructor(public store: Store<ILayerState>) {
 	}
 
-	public onCheckboxClicked(event, layer: Layer): void {
+	public onCheckboxClicked(event, layersContainer: Layer): void {
 		if (event.target.checked) {
-			this.store.dispatch(new SelectLayerAction(layer));
+			this.store.dispatch(new SelectLayerAction(layersContainer));
 		} else {
-			this.store.dispatch(new UnselectLayerAction(layer));
+			this.store.dispatch(new UnselectLayerAction(layersContainer));
 		}
+		this.collectionWithIsChecked = this.collectionWithIsChecked.map((layer) => {
```

Comment:
I think a better approach to manage the selected layers would be using a Set. You don't need another "withIsChecked" collection. Tick a checkbox of layer.id is in the set, remove/add an id from/to the set onCheckboxClicked and simply pass the set to the reducer.